framework/tst/dslabs/framework/testing/junit/BaseJUnitTest.java

```diff
@@ -381,6 +384,27 @@ private void assertGoalFound(boolean endTestOnFailure) {
             default -> "";
         });
 
+        if (GlobalSettings.startVisualization() && bfsStartState != null) {
+            // Without this, you get a single-state trace which only has the last message sent; you
```

Comment:
I actually don't understand this comment. If you just got rid of this line and wrote

```
VizClient vc = new VizClient(bfsStartState, null, true);
```

the problem would be that you're referencing a field of `BaseJUnitTest` that's about to get set to `null`. But even in that case, you should either see an NPE or get a completely working trace.

Reply:
If you have a reproducible example you want to email to me, I'd be interested to take a look.

Reply:
Ohhh..... I bet I know what's going on. You saw this right after a call to https://github.com/emichael/dslabs/blob/b63d3cf55df28535d315a12ea386d1cfbdbeac85/framework/tst/dslabs/framework/testing/search/SearchState.java#L473 didn't you? You should still have seen the previous states down in the bottom of the screen, but you wouldn't have seen any messages. Or, it's possible that something in Oddity didn't like that messages disappeared. In any case, if you can repro, I'd like to take a look.

The right thing to do long term is probably to change the interface to `dropPendingMessages` so that it returns a new state and doesn't actually modify the current one.

Reply:
Resolved over email -- click "Debugger > Debug System With Trace", not "Debugger > Debug System".